pkg/apiserver/storage/vpc_inventory.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"fmt"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/apis/meta/internalversion"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+
+	antreastorage "antrea.io/antrea/pkg/apiserver/storage"
+	"antrea.io/antrea/pkg/apiserver/storage/ram"
+	"antrea.io/antrea/pkg/controller/types"
+	runtimev1alpha1 "antrea.io/nephe/apis/runtime/v1alpha1"
+)
+
+const (
+	VpcIndexerByAccountNameSpacedName = "namespace.accountname"
+	VpcIndexerByVpcNamespacedName     = "namespace.vpcname"
+	VpcIndexerByNamespace             = "vpc.namespace"
+	VpcIndexerByNamespacedRegion      = "namespace.region"
+
+	VpcLabelAccountName = "account-name"
+	VpcLabelRegion      = "region"
+)
+
+// vpcInventoryEvent implements storage.InternalEvent.
+type vpcInventoryEvent struct {
+	// The current version of the stored VPC.
+	CurrObject *runtimev1alpha1.Vpc
+	// The previous version of the stored VPC.
+	PrevObject *runtimev1alpha1.Vpc
+	// The key of this VPC.
+	Key             string
+	ResourceVersion uint64
+}
+
+// keyAndSpanSelectFunc returns whether the provided selectors matches the key and/or the nodeNames.
+func keyAndSpanSelectFunc(selectors *antreastorage.Selectors, key string, obj interface{}) bool {
+	// If Key is present in selectors, the provided key must match it.
+	if selectors.Key != "" && key != selectors.Key {
+		return false
+	}
+	// If nodeName is present in selectors's Field selector, the provided nodeNames must contain it.
+	if nodeName, found := selectors.Field.RequiresExactMatch("nodeName"); found {
```

Comment:
I dont think this is required.. You can try removing it and reverify

Reply:
deleted